### PR TITLE
Safari Performance Test Integration 2/2

### PR DESCRIPTION
**Pull request by afterxleep**
Task/Issue URL: https://app.asana.com/1/137249556945/task/1211159858495832
Tech Design URL: N/A
CC: N/A

### Description
Adds results visualization and comparison features for Safari performance testing. Builds on Phase 1 by introducing  a metrics comparison UI, Safari results parsing, and browser comparison analytics.

It also implements a few fixes for consistency:
- Use new tabs for DDG tests
- Split & Clean up long methods

### Testing Steps

- Make sure you have node.js installed
- Go to Safari > Settings > Developer and check "Enable Remote Automation"
- Close Safari

Then...

1. Open any website in DuckDuckGo macOS browser
2. Select "Test Current Site Performance (Safari)" from debug menu
3. Run a test with Safari
4. Verify the new comparison UI displays metrics side-by-side
5. Check that metrics calculations are accurate (percentiles, medians, etc.)
6. Verify the comparison metric boxes show DuckDuckGo vs Safari performance
7. Test with multiple iterations to ensure statistics are calculated correctly

### Impact and Risks
**Impact Level: Low**

#### What could go wrong?
Metrics calculation errors could show incorrect performance comparisons. All calculations include proper validation and edge case handling.

### Quality Considerations
- Tests updated for new components
- SafariResultsParser handles various result formats
- Comparison metrics include proper statistical calculations (P95, median, etc.)
- UI handles empty states and error scenarios
- All new code follows existing patterns from Phase 1

### Notes to Reviewer
Phase 2 of Safari test integration. Completes the feature by adding results visualization and comparison capabilities promised in Phase 1.

---

> [!NOTE]
> No changes in this PR (empty diff).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 29d6ab63ffb7ccd4e68b6e5f4890d3f9986d6508. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: macOS/LocalPackages/PerformanceTest/Sources/PerformanceTest/UI/PerformanceTestViewModel.swift [hunk 1]
```diff
@@ -71,34 +89,115 @@ final class PerformanceTestViewModel: ObservableObject {
...
+                    url: url,
+                    iterations: selectedIterations + 1, // +1 for warm-up run
+                    timeout: PerformanceTestConstants.TestConfig.testTimeout
+                )
+            }
+        }
+
+        // Clear webView references to allow proper deallocation
+        webView = nil
+        self.duckDuckGoTester = nil
```
[the remark this answers is not in the document]
Comment (afterxleep) [reply macOS/LocalPackages/PerformanceTest/Sources/PerformanceTest/UI/PerformanceTestViewModel.swift]: The nil assignment is SAFE and INTENTIONAL - happens AFTER DuckDuckGo test completes, before Safari test - Allows proper memory cleanup between tests

File: macOS/DuckDuckGo/MainWindow/MainViewController.swift [hunk 2]
```diff
@@ -851,27 +851,39 @@ extension MainViewController {
...
         // Use the package to handle everything
-        let windowController = PerformanceTestWindowController(webView: currentTab.webView)
-        windowController.showWindow(nil)
-    }
+        let windowController = PerformanceTestWindowController(
+            webView: currentTab.webView,
+            createNewTab: { @MainActor [weak self] in
+                guard let self = self else {
+                    fatalError("MainViewController deallocated during performance test")
+                }
```
[the remark this answers is not in the document]
Comment (afterxleep) [reply macOS/DuckDuckGo/MainWindow/MainViewController.swift]: wut?   The constant EXISTS at PerformanceTestConstants.swift:136: static let iqr = "iqr"

File: macOS/DuckDuckGo/MainWindow/MainViewController.swift [hunk 3]
```diff
@@ -851,27 +851,45 @@ extension MainViewController {
...
-              let url = currentTab.url else {
-            let alert = NSAlert()
-            alert.messageText = "No Active Page"
-            alert.informativeText = "Please navigate to a webpage first to test its performance with Safari."
-            alert.alertStyle = .informational
-            alert.addButton(withTitle: "OK")
-            alert.runModal()
-            return
-        }
+                // Create a new tab with duckduckgo.com for JS warmup
```
Comment (afterxleep) [inline macOS/DuckDuckGo/MainWindow/MainViewController.swift:862]: We're now using new tabs
